extensions/guacamole-vault/modules/guacamole-vault-ksm/src/main/java/org/apache/guacamole/vault/ksm/secret/KsmRecordService.java

```diff
@@ -112,6 +112,29 @@ private <T> T getSingleValue(List<T> values) {
 
     }
 
+    /**
+     * Returns the single value stored in the given list of strings. If the
```

Comment:
Shouldn't this comment say that `null` will be returned if the string is `null`? That's the whole point of this fix but it's left unclear in the block comments for these functions.

Reply:
To clarify, the point of the fix is that `null` should be returned if the string is the _empty string_, but yes it does say that. See immediately after the sentence in question:

> ... If the list is empty, contains multiple values, or contains only a single empty string, null is returned.

It's also listed in the `@return`:

```
     * @return
     *     The single value stored in the given list, or null if the list is
     *     empty, contains multiple values, or contains only a single empty
     *     string.
     */
```

Reply:
Ah, I see I had it backwards, I thought the fix was that it was now returning `null`, not that it is now returning an empty string. Gotcha.

That said, I still don't see where it says it will return `null` if the string is `null`. Your quoted text says it will return `null` in the following cases:

- The list is empty
- The list contains multiple values
- The list contains only a single empty string

My point is that it never says that it will return `null` if the string is `null`.

Reply:
I guess the logical assumption would be that it would return `null` in that case even if we don't explicitly state it, but if we're gonna list out all these other possibilities, I don't see why we shouldn't list this one as well.

Reply:
My thinking was that the documentation was already clear on that point. If the list contains a single `null`, then that's what's documented as returned ("the single value stored in the given list"), with the only exceptions to that rule being the case listed ("a single empty string").

Reply:
Looking at `Map`, where `get()` returns `null` if there is no value stored for a particular key, the Java folks feel it necessary to note this ambiguous case:

> A return value of null does not necessarily indicate that the map contains no mapping for the key; it's also possible that the map explicitly maps the key to null.

How about:

> Returns the single value stored in the given list of strings. If the list is empty, contains multiple values, contains only a single empty string, or contains only a single null value, null is returned.

?

Reply:
I've pushed commit 837a036 to add a clarifying note regarding the behavior of `null` values to each of the "getSingle" functions.

Reply:
LGTM